internal/mongodoc/resource.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package mongodoc
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"gopkg.in/errgo.v1"
+	"gopkg.in/juju/charm.v6-unstable"
+	"gopkg.in/juju/charm.v6-unstable/resource"
+	"gopkg.in/juju/charmrepo.v2-unstable/csclient/params"
+	"gopkg.in/mgo.v2/bson"
+)
+
+// NewResourceQuery formats the provided details into a mongo query for
+// the identified resource.
+func NewResourceQuery(cURL *charm.URL, resName string, revision int) bson.D {
+	copied := *cURL
+	cURL = &copied
+	cURL.Revision = -1
+	cURL.Series = ""
+	query := bson.D{
+		{"unresolved-charm-url", cURL},
+		{"name", resName},
+	}
+	if revision >= 0 {
+		query = append(query, bson.DocElem{"revision", revision})
+	}
+	return query
+}
+
+// Resource holds the in-database representation of a charm resource
+// at a particular revision.
+type Resource struct {
+	// CharmURL identifies the unresolved charm associated with this
+	// resource.
+	CharmURL *charm.URL `bson:"unresolved-charm-url"`
```

Comment:
Could you add a comment saying what unresolved means in this context? 

Also, if it's a URL without a revision that's fine. If it's a url without a series or revision then I think I'd prefer it be called BaseURL to be consistent with the rest of the database.